### ISSUE TITLE
Options for `lines-before-comment` to ignore the beginning of a class, function or other block

### ISSUE THREAD

**Issue by ehoogeveen-medweb** [gajus/eslint-plugin-jsdoc#1296]
## Motivation

For example:
```js
class MyClass {
  /**
   * Description of member function.
   */
  outerFunc() {
    /**
     * Description of inner function.
     */
    innerFunc() { }

    return innerFunc;
  }
}
```
Compare the options in the ESLint Stylistic rule [lines-around-comment](https://eslint.style/rules/default/lines-around-comment) and the [typescript variant](https://eslint.style/rules/ts/lines-around-comment).

## Current behavior

Warns about jsdoc comments immediately after the start of a class/function/block.

## Desired behavior

Options not to warn about those positions.

**Comment by JoshuaKGoldberg** [comment]
IMO this is a bug, or at least an option that should be enabled by default. It's very rare and nitpicky that projects would enforce the start of the first JSDoc comment in a `{}` block need a blank line above it.
